components/eamxx/src/physics/mam/eamxx_mam_aci_functions.hpp

```diff
@@ -0,0 +1,726 @@
+  int lmassptr_amode[maxd_aspectype][ntot_amode]  = {};
+  int numptr_amode[ntot_amode]                    = {};
+  int mam_idx[ntot_amode][nspec_max]              = {};
+  int mam_cnst_idx[ntot_amode][nspec_max]         = {};
+
+  Real specdens_amode[maxd_aspectype] = {};
+  Real spechygro[maxd_aspectype]      = {};
+  Real exp45logsig[ntot_amode] = {}, alogsig[ntot_amode] = {},
+       num2vol_ratio_min_nmodes[ntot_amode] = {},
+       num2vol_ratio_max_nmodes[ntot_amode] = {};
+  Real aten                                 = 0;
+  mam4::ndrop::get_e3sm_parameters(nspec_amode, lspectype_amode, lmassptr_amode,
+                                   numptr_amode, specdens_amode, spechygro,
+                                   mam_idx, mam_cnst_idx);
+  mam4::ndrop::ndrop_init(exp45logsig, alogsig, aten, num2vol_ratio_min_nmodes,
+                          num2vol_ratio_max_nmodes);
+  //---------------------------------------------------------------------------
+  //---------------------------------------------------------------------------
+
+  Kokkos::parallel_for(
+      team_policy, KOKKOS_LAMBDA(const haero::ThreadTeam &team) {
+        const int icol = team.league_rank();
+        // for (int icol=0; icol<5; ++icol){
+        mam4::ndrop::View1D raercol_cw_view[mam4::ndrop::pver][2];
+        mam4::ndrop::View1D raercol_view[mam4::ndrop::pver][2];
+        for(int i = 0; i < mam4::ndrop::pver; ++i) {
+          for(int j = 0; j < 2; ++j) {
+            raercol_cw_view[i][j] = ekat::subview(loc_raercol_cw[i][j], icol);
+            raercol_view[i][j]    = ekat::subview(loc_raercol[i][j], icol);
+          }
+        }
+        mam4::ColumnView qqcw_view[mam4::ndrop::ncnst_tot];
+        for(int i = 0; i < mam4::ndrop::ncnst_tot; ++i) {
+          qqcw_view[i] = ekat::subview(qqcw_fld_work_loc[i], icol);
+        }
+        mam4::ColumnView ptend_q_view[mam4::aero_model::pcnst];
+        for(int i = 0; i < mam4::aero_model::pcnst; ++i) {
+          ptend_q_view[i] = ekat::subview(loc_ptend_q[i], icol);
+        }
+        mam4::ColumnView coltend_view[mam4::ndrop::ncnst_tot],
+            coltend_cw_view[mam4::ndrop::ncnst_tot];
+        for(int i = 0; i < mam4::ndrop::ncnst_tot; ++i) {
+          coltend_view[i]    = ekat::subview(loc_coltend[i], icol);
+          coltend_cw_view[i] = ekat::subview(loc_coltend_cw[i], icol);
+        }
+
+        // To construct state_q, we need fields from Prognostics (aerosols)
+        //  and Atmosphere (water species such as qv, qc etc.)
+
+        // get prognostics
+        mam4::Prognostics progs_at_col = aerosols_for_column(dry_aero, icol);
+
+        // get atmospheric quantities
+        haero::Atmosphere haero_atm =
+            atmosphere_for_column(dry_atmosphere, icol);
+
+        // Construct state_q (interstitial) and qqcw (cloud borne) arrays
+        Kokkos::parallel_for(
+            Kokkos::TeamThreadRange(team, 0u, mam4::ndrop::pver),
+            [=](int klev) {
```

Comment:
Prefer `[&]` over `[=]` for inner loops lambdas. Also, you should use `TeamVectorRange` rather than `TeamThreadRange`. The former ensures each iteration is handled by a different thread in a warp, while the latter would be problematic if by chance the team policy changes and has a vector_length!=1.

Reply:
Done